Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp

```diff
@@ -3964,3 +3983,46 @@ void webkit_settings_set_media_content_types_requiring_hardware_support(WebKitSe
     priv->mediaContentTypesRequiringHardwareSupport = mediaContentTypesRequiringHardwareSupportString.utf8();
     g_object_notify_by_pspec(G_OBJECT(settings), sObjProperties[PROP_MEDIA_CONTENT_TYPES_REQUIRING_HARDWARE_SUPPORT]);
 }
+
+/**
+ * webkit_settings_get_enable_media_session:
+ * @settings: a #WebKitSettings
+ *
+ * Get the [property@Settings:enable-media-session] property.
+ *
+ * Returns: %TRUE If MediaSession support is enabled or %FALSE otherwise.
+ *
+ * Since: 2.40
+ */
+gboolean webkit_settings_get_enable_media_session(WebKitSettings* settings)
+{
+    g_return_val_if_fail(WEBKIT_IS_SETTINGS(settings), FALSE);
+
+    return settings->priv->preferences->mediaSessionEnabled();
+}
+
+/**
+ * webkit_settings_set_enable_media_session:
+ * @settings: a #WebKitSettings
+ * @enabled: Value to be set
+ *
+ * Set the [property@Settings:enable-media-session] property.
+ *
+ * Since: 2.40
+ */
+void webkit_settings_set_enable_media_session(WebKitSettings* settings, gboolean enabled)
+{
+    g_return_if_fail(WEBKIT_IS_SETTINGS(settings));
+
+    WebKitSettingsPrivate* priv = settings->priv;
+    bool currentValue = priv->preferences->mediaSessionEnabled();
+    if (currentValue == enabled)
+        return;
+
+#if ENABLE(MEDIA_SESSION)
+    priv->preferences->setMediaSessionEnabled(enabled);
+    priv->preferences->setMediaSessionCoordinatorEnabled(enabled);
+    priv->preferences->setMediaSessionPlaylistEnabled(enabled);
+#endif
+    g_object_notify_by_pspec(G_OBJECT(settings), sObjProperties[PROP_ENABLE_MEDIA_SESSION]);
```

Comment:
This will notify twice if setting the property directly, but I notice all other settings have the same bug, so no need to change anything here: might as well be consistent with the other settings.

If we ever find a 25 hour day, we should do an audit to see where else we are missing G_PARAM_EXPLICIT_NOTIFY.